framework/plugins/plugin.go

```diff
@@ -159,6 +160,51 @@ func resolvePlugins(pluginsDir, assetsDir, downloadsDir string, osArch osarch.OS
 	return nil, errors.New(strings.Join(errStringsParts, "\n"+strings.Repeat(" ", pluginsinternal.IndentSpaces)))
 }
 
+func groupByProductAndGroup(plugins []godellauncher.SinglePluginParam) map[string][]godellauncher.SinglePluginParam {
+	pluginByGroupAndProduct := make(map[string][]godellauncher.SinglePluginParam)
+	for _, currPlugin := range plugins {
+		groupKey := currPlugin.LocatorWithChecksums.GroupAndProductString()
+		_, ok := pluginByGroupAndProduct[groupKey]
+		if !ok {
+			pluginByGroupAndProduct[groupKey] = make([]godellauncher.SinglePluginParam, 0)
+		}
+		singlePluginList := pluginByGroupAndProduct[groupKey]
+		pluginByGroupAndProduct[groupKey] = append(singlePluginList, currPlugin)
+	}
+	return pluginByGroupAndProduct
+}
+
+// dedupePlugins will remove any plugins that added by the configuration provider iff there is a plugin
```

Comment:
Typos in comment

Reply:
Fixed "pluging". "iff" was intentional to be "if and only if"